backend/src/routes/classroom.ts

```diff
@@ -1,12 +1,62 @@
-import { Request, Response, Router } from 'express';
+import { NextFunction, Request, Response, Router } from 'express';
 import Classroom from 'models/classroom';
+import { makeError } from 'helpers/error';
+import { NOT_FOUND } from 'http-status';
 
 const router = Router();
 
-router.get('/', async (req: Request, res: Response) => {
-  const classrooms = await Classroom.findAll();
+router.get('/', async (req: Request, res: Response, next: NextFunction) => {
+  await Classroom.findAll()
+    .then(classrooms => res.json(classrooms))
+    .catch(error => next(error));
+});
+
+router.get('/:id', async (req: Request, res: Response, next: NextFunction) => {
+  await Classroom.findByPk(req.params.id)
+    .then(classroom => {
+      if (!classroom) {
+        return next(makeError('Classroom with that id not found.', NOT_FOUND));
+      }
+      res.json(classroom);
+    })
```

Comment:
Nitpick:
```js
if (classroom) return res.json(classroom);
return next(makeError('Classroom with that id not found.', NOT_FOUND));
```

Reply:
This was my original code but I was persuaded into this way. I prefer ternary usually when there are two states, but this is my preferred way when working when somebody has to read it 😂 It's relative but yeah when you have two possible states it better to check positives first because you get rid of negation step. But I would like to talk about why you prefer this way  in person 😊.